File: MonthlyChallenges/AprilChallenge/01-reversestring.cs
// LeetCode #344 - Easy
// 04-01-2022
// 274ms Faster Than 74.95%
// 48MB Less Than 57.31%
using System;

namespace LeetCode
{
    public class Program
    {
        public static void Main(string[] args)
        {
            Solution sol = new Solution();
        }
    }

    public class Solution
    {
        public void ReverseString(char[] s)
        {
            int midpoint = s.Length / 2;

            int k = s.Length - 1;
            for (int i = 0; i < midpoint; i++)
            {
                char temp = s[i];
                s[i] = s[k];
                s[k] = temp;
                k--;
            }
        }
    }
}

File: MonthlyChallenges/AprilChallenge/01-reversestring.cpp
// LeetCode #344 - Easy
// 04-01-2022
// 32ms Faster Than 45.20%
// 23.2MB Less Than 76.65%
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    void reverseString(vector<char>& s)
    {
        int midpoint = s.size() / 2;

        int k = s.size()-1;
        for (int i = 0; i < midpoint; i++)
        {
            char temp = s[i];
            s[i] = s[k];
            s[k] = temp;
            k--;
        }
    }
};